test/multilayer_join_test.cpp

```
@@ -0,0 +1,105 @@
+  struct input_type_2 {
+    int value;
+  };
+
+  template <typename T>
+  product_specification spec(char const* creator, char const* suffix)
+  {
+    return {algorithm_name{creator}, identifier{suffix}, make_type_id<T>()};
+  }
+
+  template <typename T>
+  product_selector selector(char const* creator, char const* suffix)
```

Cppcheck (2.21.0)

[style] The function 'selector' is never used.

(unusedFunction)